python/plugins/processing/gui/AlgorithmDialog.py

```diff
@@ -110,8 +112,12 @@ def getParameterValues(self):
                     if self.mainWidget().checkBoxes[param.name()].isChecked():
                         dest_project = QgsProject.instance()
 
-                value = self.mainWidget().outputWidgets[param.name()].getValue()
-                if value and isinstance(value, QgsProcessingOutputLayerDefinition):
+                widget = self.mainWidget().outputWidgets[param.name()]
+                value = widget.getValue()
+                if value and not isinstance(param, (QgsProcessingParameterFolderDestination,
```

Comment:
This probably should be `if value is not None and not isinstance(value, NULL)...`

Reply:
This comes from DestinationSelectionPanel.accept(), we do not want to create a QgsProcessingOutputLayerDefinition in case of QgsProcessingParameterFolderDestination and QgsProcessingParameterFileDestination, could be more clear with:
```
if ... and isintance(param, (
    QgsProcessingParameterFeatureSink,
    QgsProcessingParameterRasterDestination, 
    QgsProcessingParameterVectorDestination):
```